src/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, { PropsWithChildren } from "react";
+import styles from "./Toggle.module.css";
+
+type ToggleProps = {
+  className?: string;
+} & React.ComponentPropsWithoutRef<"input">;
+
+/**
+ * Toggle form control
+ */
```

Comment:
I might be misunderstanding something here but personally I'd expect to see more information about what exactly this is ("React component which renders a checkbox control for a form")  and how it can be used by an application:

What are the supported properties? Examples of common ones (`disabled`, `checked`) would be helpful. How does the application read the state?

Reply:
I agree, the description of the component could be much better here! I will update that. 

I am not quite sure we should write down the supported properties here, they're defined by the TypeScript type just above, and would be concerned these two things go out of sync.

The discovery of the component variants will be achieved through Storybook stories in the long term, https://compound.element.io/?path=/story/compound-web_alert--success (see the right panel as an example)
But we are yet to formalise an approach here